db_test.go

```diff
@@ -1303,6 +1303,114 @@ func TestInitializeHeadTimestamp(t *testing.T) {
 	})
 }
 
+func TestDB_LabelNames(t *testing.T) {
+	tests := []struct {
+		sampleLabels1 [][2]string // For checking head and disk.
+		// sampleLabels2 should have atleast 1 label name which is not there in sampleLabels1.
+		sampleLabels2 [][2]string // For check head+disk.
+		exp1          []string    // after adding sampleLabels1.
+		exp2          []string    // after adding sampleLabels1 and sampleLabels2.
+
+		ms    []labels.Matcher
+		expMs []string // after adding sampleLabels1 and sampleLabels2.
+	}{
+		{
+			sampleLabels1: [][2]string{
+				[2]string{"name1", ""},
+				[2]string{"name3", ""},
+				[2]string{"name2", ""},
+			},
+			sampleLabels2: [][2]string{
+				[2]string{"name4", ""},
+				[2]string{"name1", ""},
+			},
+			exp1:  []string{"name1", "name2", "name3"},
+			exp2:  []string{"name1", "name2", "name3", "name4"},
+			ms:    []labels.Matcher{labels.NewEqualMatcher("", "name2")},
+			expMs: []string{"name2"},
+		},
+		{
+			sampleLabels1: [][2]string{
+				[2]string{"name2", ""},
+				[2]string{"name1", ""},
+				[2]string{"name2", ""},
+			},
+			sampleLabels2: [][2]string{
+				[2]string{"name6", ""},
+				[2]string{"name0", ""},
+			},
+			exp1: []string{"name1", "name2"},
+			exp2: []string{"name0", "name1", "name2", "name6"},
+		},
+	}
+
+	m, err := labels.NewRegexpMatcher("", ".*2|.*6")
+	testutil.Ok(t, err)
+	tests[1].ms = []labels.Matcher{m}
+	tests[1].expMs = []string{"name2", "name6"}
```

Comment:
instead of this you can add it in the test cases as:
```
exp1:  []string{"name1", "name2"},
exp2:  []string{"name0", "name1", "name2", "name6"},
ms:    []labels.Matcher{labels.NewMustRegexpMatcher("", ".*2|.*6")},
expMs: []string{"name2", "name6"},
```